File: src/Bionic.Bridge.Capacitor.JS/src/SplashScreenBridge.ts
import {Plugins, SplashScreenHideOptions, SplashScreenShowOptions, StatusBarStyle} from "@capacitor/core";

const {SplashScreen} = Plugins;

interface ISplashScreen {
  show(options: SplashScreenShowOptions): Promise<void>;
  hide(options: SplashScreenHideOptions): Promise<void>;
}

export class SplashScreenBridge implements ISplashScreen {
  async show(options: SplashScreenShowOptions): Promise<void> {
    if (!SplashScreen) return Promise.reject("SplashScreen not available.");
    return await SplashScreen.show(options);
  }

  async hide(options: SplashScreenHideOptions): Promise<void> {
    if (!SplashScreen) return Promise.reject("SplashScreen not available.");
    return await SplashScreen.hide(options);
  }
}

File: src/Bionic.Bridge.Capacitor/SplashScreenBridge.cs
using System.Threading.Tasks;
using Microsoft.JSInterop;

namespace Bionic.Bridge.Capacitor {
    public static class SplashScreenBridge {
        private const string SplashScreenShowFunctionName = "BionicBridge.Capacitor.SplashScreen.show";
        private const string SplashScreenHideFunctionName = "BionicBridge.Capacitor.SplashScreen.hide";

        public static Task Show(SplashScreenShowOptions options = null) =>
            JSRuntime.Current.InvokeAsync<object>(SplashScreenShowFunctionName, options);

        public static Task Hide(SplashScreenHideOptions options = null) =>
            JSRuntime.Current.InvokeAsync<object>(SplashScreenHideFunctionName, options);
    }

    public class SplashScreenShowOptions {
        public bool autoHide;
        public long fadeInDuration;
        public long fadeOutDuration;
        public long showDuration;
    }

    public class SplashScreenHideOptions {
        public long fadeOutDuration;
    }
}